src/ioctl.rs

```diff
@@ -182,12 +348,68 @@ pub unsafe fn ioctl_with_mut_ref<F: AsRawFd, T>(fd: &F, req: c_ulong, arg: &mut
     )
 }
 
-/// Run an ioctl with a raw pointer.
+/// A wrapper of unsafe [libc::ioctl] with a raw pointer.
+///
+/// # Arguments
+///
+/// * `fd`: an open file descriptor.
+/// * `req`: a device-dependent request code.
+/// * `arg`: a raw pointer passed to ioctl.
+///
+/// # Safety
+///
+/// The caller should ensure to pass a valid file descriptor and have the
+/// return value checked.
+///
+/// # Examples
+///
+/// ```
+/// extern crate vmm_sys_util;
+/// #
+/// # use std::fs::File;
+/// # use std::os::raw::c_ulong;
+/// # use std::os::unix::io::FromRawFd;
+/// use vmm_sys_util::ioctl::ioctl_with_ptr;
+///
+/// const TUNSETOFFLOAD: c_ulong = 0x4004_54d0;
+/// let bytes : Vec<u8> = vec![0; 10];
+/// let file = unsafe { File::from_raw_fd(0) };
+///
+/// let ret = unsafe { ioctl_with_ptr(&file, TUNSETOFFLOAD, bytes.as_ptr()) };
+/// ```
 pub unsafe fn ioctl_with_ptr<F: AsRawFd, T>(fd: &F, req: c_ulong, arg: *const T) -> c_int {
     libc::ioctl(fd.as_raw_fd(), req as IoctlRequest, arg as *const c_void)
 }
 
-/// Run an ioctl with a mutable raw pointer.
+/// A wrapper of unsafe [libc::ioctl] with a mutable raw pointer.
+///
+/// # Arguments
+///
+/// * `fd`: an open file descriptor.
```

Comment:
The `fd` is not any opened file descriptor. It is the open file descriptor corresponding to the device on which we want to call the ioctl. Please update the documentation for all the other wrappers over ioctls as well.

Reply:
Got it.